### ISSUE TITLE
Problems with new API endpoints in the nextjs starter project and getStaticPaths

### ISSUE THREAD

**Issue by FelixButzbach** [zeit/now#6804]
TL;DR
When using `getStaticPaths` (and `getStaticProps`) the API endpoints that are called must already exist. They cannot be deployed with the same commit, as next will try to fetch the data during the build and this will fail and thus nothing will be deployed to vercel. When using the internal api, a workaround is to first deploy the api endpoint and in a second commit the pages that consume the endpoint.

Now the hole story and why I actually opened the issue:

For a long time I wanted to try nextjs and for a quick test for a possible new side project I thought it would be a good idea to give it a try. 
With the apparently super easy hosting and deploy via vercel I felt motivated and followed all the instructions and steps from https://vercel.com/new and it really was easy.
Then I tried to add a custom api endpoint and use it to do something similar like written here [https://nextjs.org/docs/basic-features/data-fetching](https://nextjs.org/docs/basic-features/data-fetching) as my sideproject would need several very simple. static pages that should be generated by some cms in the future.
So I created the endpoints and used some static json file to simulate a database and everything worked as expected. But only on localhost. When pushing to the repo and waiting for the build the problems started. After about 6 hours (within two days) I almost quit the complete project but for sure nextjs and vercel.
But in a 'aha moment' in the shower I had one last idea and it worked.

The problem was that while building and deploying, nextjs already tried to fetch the data from the new endpoint but the new endpoint was not yet deployed and thus the build aborted. So the solution was to comment out the `getStaticPaths` and `getStaticProps` functions and all related code, deploy this to vercel and then uncomment both functions again. The next deploy worked without problems, because now the new api endpoints were there and the data could be fetched...

I don't know if I did something wrong here or if I oversaw some important part of the documentation, but I think I pretty much just followed the getting started tutorial and ran into a pretty unobvious problem. And I wonder if I always will have to first deploy the api endpoint before deploying code that consumes it (assuming I stick with the integrated api).

Anyways, just leaving this here as I actually like all the ideas behind vercel and nextjs and I imagine that this type of problem might piss off others before actually getting to write any real code.
Or maybe I was just dump and it is already written somewhere super obvious...

**Comment by leerob** [comment]
Hey! We recommend directly calling the logic that is inside your API routes from `getStaticProps` and friends. This is because both run on the server, so there's no need to make an extra network hop to call the API route.

Let me know if you have further questions 🙏
